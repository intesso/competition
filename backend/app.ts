import Koa from 'koa'
import Router from 'koa-router'
import cors from '@koa/cors'
import bodyParser from 'koa-bodyparser'
import { inputValidation } from './api/inputValidation'
import { calculationController } from './src/calculation/controller'

export const app = new Koa()
app.use(bodyParser())
app.use(cors())

const router = new Router()

router.get('/', (ctx: Koa.Context) => {
  ctx.body = 'Hello Koa'
})

router.get('/pets', inputValidation.validate, async (ctx) => {
  ctx.status = 200
  ctx.body = { result: 'OK' }
})

app.use(async (ctx, next) => {
  try {
    return await next()
  } catch (err) {
    if (err instanceof inputValidation.InputValidationError) {
      ctx.status = 400
      ctx.body = err.errors
    }
    throw err
  }
})

app.use(router.routes())
app.use(calculationController.routes())
