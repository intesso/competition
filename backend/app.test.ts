import request from 'supertest'
import { app } from './app'

test('Hello world works', async () => {
  const response = await request(app.callback()).get('/')
  expect(response.status).toBe(200)
  expect(response.text).toBe('Hello World')
  expect(response.text).toMatchSnapshot()
})
