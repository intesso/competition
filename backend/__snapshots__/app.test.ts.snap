// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello world works 1`] = `"Hello World"`;
